### PR TITLE
[FIX] Created TVTKBaseInterface. Should fix #1218

### LINKED ISSUE #1218: AttributeError: 'PolyDataWriter' object has no attribute 'set_input_data'
```
> DISPLAY= PYTHONPATH=/home/yoh/proj/nipy/nipy-suite/build-area/nipype-0.11.0/debian/python-nipype/usr/lib/python2.7/site-packages MPLCONFIGDIR=/home/yoh/proj/nipy/nipy-suite/build-area/nipype-0.11.0/build xvfb-run -s "-screen 0 1024x768x24 -ac  +render -noreset" nosetests nipype.algorithms.tests.test_mesh_ops
Xlib:  extension "RANDR" missing on display ":99".
EE
======================================================================
ERROR: Failure: AttributeError ('PolyDataWriter' object has no attribute 'set_input_data'
Interface ComputeMeshWarp failed to run. )
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 251, in generate
    for test in g():
  File "/usr/lib/python2.7/dist-packages/numpy/testing/decorators.py", line 153, in skipper_gen
    for x in f(*args, **kwargs):
  File "/home/yoh/proj/nipy/nipy-suite/build-area/nipype-0.11.0/debian/python-nipype/usr/lib/python2.7/site-packages/nipype/algorithms/tests/test_mesh_ops.py", line 35, in test_ident_distances
    res = dist_ident.run()
  File "/home/yoh/proj/nipy/nipy-suite/build-area/nipype-0.11.0/debian/python-nipype/usr/lib/python2.7/site-packages/nipype/interfaces/base.py", line 1033, in run
    runtime = self._run_wrapper(runtime)
  File "/home/yoh/proj/nipy/nipy-suite/build-area/nipype-0.11.0/debian/python-nipype/usr/lib/python2.7/site-packages/nipype/interfaces/base.py", line 983, in _run_wrapper
    runtime = self._run_interface(runtime)
  File "/home/yoh/proj/nipy/nipy-suite/build-area/nipype-0.11.0/debian/python-nipype/usr/lib/python2.7/site-packages/nipype/algorithms/mesh.py", line 283, in _run_interface
    writer.set_input_data(out_mesh)
AttributeError: 'PolyDataWriter' object has no attribute 'set_input_data'
Interface ComputeMeshWarp failed to run. 
-------------------- >> begin captured logging << --------------------
rdflib: INFO: RDFLib Version: 4.1.2
interface: INFO: Redirecting X to :1099
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: Failure: AttributeError ('PolyDataWriter' object has no attribute 'set_input_data')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/loader.py", line 251, in generate
    for test in g():
  File "/usr/lib/python2.7/dist-packages/numpy/testing/decorators.py", line 153, in skipper_gen
    for x in f(*args, **kwargs):
  File "/home/yoh/proj/nipy/nipy-suite/build-area/nipype-0.11.0/debian/python-nipype/usr/lib/python2.7/site-packages/nipype/algorithms/tests/test_mesh_ops.py", line 62, in test_trans_distances
    writer.set_input_data(vtk1)
AttributeError: 'PolyDataWriter' object has no attribute 'set_input_data'

----------------------------------------------------------------------
Ran 2 tests in 0.626s

FAILED (errors=2)
```

which version of tvtk (mayavi2) do you need? we have 4.3.1-4.1 on debian sid/stretch atm

### PR DESCRIPTION
**Pull request by oesteban**

### CODE REVIEW

File: nipype/algorithms/mesh.py [hunk 1]
```diff
@@ -22,9 +22,40 @@
 from ..interfaces.base import (BaseInterface, traits, TraitedSpec, File,
                                BaseInterfaceInputSpec)
 from warnings import warn
+
+
+have_tvtk = False
+
+try:
+    from tvtk.api import tvtk
```
Comment (satra) [inline nipype/algorithms/mesh.py]: the problem is that this import results in a sys exit call, if tvtk exists and DISPLAY is unavailable. i couldn't figure out a way to trap it. in that setting it doesn't generate an ImportError. it just plain kills the interpreter.

Comment (oesteban) [inline nipype/algorithms/mesh.py]: Yep. A fix: https://github.com/nipy/nipype/issues/1178#issuecomment-140833948
And a related question: https://github.com/nipy/nipype/issues/972#issuecomment-140835937

**Comment by chrisfilo** [comment]
This might be a better way of doing this: https://github.com/nipy/nipype/blob/c31a5a819137cbef15f55f7d0a77a65d6a37a88f/nipype/interfaces/ants/base.py#L63

**Comment by oesteban** [comment]
Agreed, but that is a CommandLine interface, so it has it's own environment and we can handle it. In this case, as it is a BaseInterface, the system-wide environment is used.

**Comment by oesteban** [comment]
@yarikoptic @satra @chrisfilo I think this side of the problem is solve, this is ready to merge.

**Pull request merged**